src/app/core/entity-components/entity-utils/entity-form.service.ts

```diff
@@ -0,0 +1,103 @@
+import { Injectable } from "@angular/core";
+import { FormBuilder, FormGroup, Validators } from "@angular/forms";
+import { FormFieldConfig } from "../entity-details/form/FormConfig";
+import { Entity } from "../../entity/entity";
+import { EntityMapperService } from "../../entity/entity-mapper.service";
+import { EntitySchemaService } from "../../entity/schema/entity-schema.service";
+
+@Injectable({
+  providedIn: "root",
+})
+export class EntityFormService {
+  constructor(
+    private fb: FormBuilder,
+    private entityMapper: EntityMapperService,
+    private entitySchemaService: EntitySchemaService
+  ) {}
+
+  public extendFormFieldConfig(
+    formFields: FormFieldConfig[],
+    entity: Entity,
+    forTable = false
+  ) {
+    formFields.forEach((formField) => {
+      try {
+        this.addFormFields(formField, entity, forTable);
+      } catch (err) {
+        console.log(`Could not create form config for ${formField.id}: ${err}`);
```

Comment:
why should this ever throw an error?

Reply:
In case some necessary information are not available. E.g. when working with attributes that have no schema and no view or edit component is provided. The call to the entity schema service would then fail.